src/main/java/fr/zcraft/quartzlib/components/nbt/NBTType.java

```diff
@@ -190,17 +248,79 @@ public Object getData(Object nmsNbtTag) {
             return null;
         }
         try {
+            switch (getNmsTagFieldName()) {
+                case "map":
+                    //Since 1.17 "map" became "tags"
+                    //h() return an unmodifiable map
+                    return Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "h");
+
+                case "list":
+                    //We recreate the list because there are no getter for the list.
+                    //TODO check if in 1.18 a getter is added for this one.
+                    List list = new ArrayList();
+
+                    for (int i = 0; i < (int) Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "size"); i++) {
+                        //Strange thing of java a call cast int into integer resulting in the impossibility
+                        // to call the method
+                        Method method;
+
+
+                        method = nmsNbtTag.getClass().getMethod("get", int.class);
+
+
+                        list.add(method.invoke(nmsNbtTag, i));
+                        //list.add(Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "get", i));
+                        // Doesn't work see above
```

Comment:
Simplify + same as above
```suggestion
                        // Strange thing of java a call cast int into integer resulting in the impossibility
                        // to call the method
                        final Method method = nmsNbtTag.getClass().getMethod("get", int.class);
                        
                        // Cannot use Reflection.cass for the same reason as above.
                        list.add(method.invoke(nmsNbtTag, i));
```